core/src/auth_service/utility.bal

```diff
@@ -0,0 +1,105 @@
+import ballerina/config as conf;
+import ballerina/crypto;
+import ballerina/jwt;
+import ballerina/time;
+
+# The `generateJWT` will generate the JWT token for the given user credentials.
+# 
+# + credentials - User credentials.
+# + return - This function will return either JWT or an appropriate error
+function generateJWT(json credentials) returns string|error {
+
+    [string, string] [email, password] = check extractCredentials(credentials);
+    json|error authenticateUserResult = authenticateUser(email, password);
+    if (authenticateUserResult is json) {
+
+        // Get id and userType.
+        string id = check trap <string>authenticateUserResult.id;
+        string scope = check trap <string>authenticateUserResult.userType;
+
+        // keyStore setup.
+        crypto:KeyStore keyStore = {
+            path: "resources/ballerinaKeystore.p12",
+            password: conf:getAsString("KEY_STORE_PASSWORD")
+        };
+
+        jwt:JwtKeyStoreConfig keyStoreConfig = {
+            keyStore: keyStore,
+            keyAlias: "ballerina",
+            keyPassword: "ballerina"
+        };
+
+        // Set JWT header.
+        jwt:JwtHeader header = {};
+        header.alg = jwt:RS256;
+        header.typ = "JWT";
+
+        // Set JWT payload.
+        jwt:JwtPayload payload = {};
+        payload.sub = "Auth token";
+        payload.iss = "environment platform";
+        payload.jti = "100078234ba23";
+        payload.customClaims = {name: id, scope: scope};
+        payload.exp = time:currentTime().time / 1000 + conf:getAsInt("JWT_EXPIRE", 3600);
```

Comment:
Is there a reason why the expire time is provided to the **getAsInt()** method? It defies the process of getting it from the config in the first place.

Reply:
Yeah because it should be a number. When we take the numbers from the config file we have to use that method. Hope I answered your question.

Reply:
Um no, what I meant was why do we have to pass in the value 3600 when that value should be retrieved from the config?

Reply:
It is the default value. getAsInt take two parameters one is the config file const and next is the default value.

Reply:
Ah right, okay. Then there is no problem there